### ISSUE TITLE
inconsistent test results

### ISSUE THREAD

**Issue by lyw07** [learningequality/kolibri#7289]
### Observed behavior

I noticed that in Travis, the results of tests `test_deleted_resources`, `test_new_resources` and `test_update_resources` are inconsistent. They would fail sometimes, but after restarting the jobs, they would pass.

### Expected behavior

test results should be consistent

### User-facing consequences

developers need to restart Travis jobs to make the tests pass

### Errors and logs

```
_________________ ChannelMixedTestCase.test_deleted_resources __________________
1387
1388
self = <kolibri.core.content.test.test_channel_upgrade.ChannelMixedTestCase testMethod=test_deleted_resources>
1389
1390
    def test_deleted_resources(self):
1391
        resources_to_be_deleted_count = count_removed_resources(
1392
            self.content_db_path, self.channel_id
1393
        )
1394
    
1395
>       self.assertEqual(resources_to_be_deleted_count, 5)
1396
E       AssertionError: 4 != 5
1397
1398
kolibri/core/content/test/test_channel_upgrade.py:591: AssertionError
1399

___________________ ChannelMixedTestCase.test_new_resources ____________________
1402
1403
self = <kolibri.core.content.test.test_channel_upgrade.ChannelMixedTestCase testMethod=test_new_resources>
1404
1405
    def test_new_resources(self):
1406
        (
1407
            new_resource_ids,
1408
            new_resource_content_ids,
1409
            new_resource_total_size,
1410
        ) = get_new_resources_available_for_import(
1411
            self.content_db_path, self.channel_id
1412
        )
1413
    
1414
        self.assertEqual(
1415
            set(new_resource_ids),
1416
>           set(map(lambda x: x["id"], self.upgraded_channel.new_resources)),
1417
        )
1418
E       AssertionError: Items in the second set but not the first:
1419
E       '911f0394683140af9ffe0e13f66033a7'
1420
1421
kolibri/core/content/test/test_channel_upgrade.py:570: AssertionError

__________________ ChannelMixedTestCase.test_update_resources __________________
self = <kolibri.core.content.test.test_channel_upgrade.ChannelMixedTestCase testMethod=test_update_resources>
    def test_update_resources(self):
        (
            updated_resource_ids,
            updated_resource_content_ids,
            updated_resource_total_size,
        ) = get_automatically_updated_resources(self.content_db_path, self.channel_id)
    
        self.assertEqual(
            set(updated_resource_ids),
>           set(map(lambda x: x["id"], self.upgraded_channel.updated_resources)),
        )
E       AssertionError: Items in the second set but not the first:
E       'f92e802f5c1a4e2da12577a76c81a173'
kolibri/core/content/test/test_channel_upgrade.py:602: AssertionError
```

### Steps to reproduce

not sure. this seems to happen randomly in Travis

### Context

Kolibri version: release-v0.14.x

**Comment by rtibbles** [comment]
Thanks - this was apparent yesterday, but an issue to track it is helpful! I also think I've realized why it is happening, so hope to get a fix in today.

**Comment by lyw07** [comment]
Fixed in #7290
